osu.Game/Database/ArchiveModelManager.cs

```diff
@@ -635,10 +639,12 @@ private List<TFileModel> createFileInfos(ArchiveReader reader, FileStore files)
             {
                 using (Stream s = reader.GetStream(file))
                 {
+                    var fileInfo = files.Add(s);
                     fileInfos.Add(new TFileModel
                     {
                         Filename = file.Substring(prefix.Length).ToStandardisedPath(),
-                        FileInfo = files.Add(s)
+                        FileInfo = fileInfo,
+                        FileInfoID = fileInfo.ID
```

Comment:
why would this be required? I'd like to think EF should track across navigational properties if the models are attached and added to the dbset?

Reply:
Did not investigate the details, but it does not work as is.
![fileInfoIdDebug](https://user-images.githubusercontent.com/22450664/111078528-ee586300-8506-11eb-8a42-2be87f3f51c8.PNG)

Reply:
not sure how to respond to this. i do not take well to just tacking "fixes" on without any understanding as to why they help.

Reply:
I'm fine with this as long as it works. Probably just need to add a comment mentioning that this is a requirement to make 5.x work correctly (and the reasoning is unknown for now).